stdlib/src/builtin/string.mojo

```diff
@@ -1333,36 +1333,93 @@ struct String(
             substr._strref_dangerous(), start=start
         )
 
-    fn split(self, delimiter: String) raises -> List[String]:
+    @always_inline
+    fn split(
+        self, delimiter: StringRef, maxsplit: Int = Int.MAX
+    ) raises -> List[String]:
```

Comment:
Why do we need this overload? I could understand having `StringRef.split` first, and maybe delegating to that?

Reply:
From my testing implicit conversion between `StringLiteral` and `Optional[String]` doesn't work so any invocations like
`String.split(",")` must become `String.split(String(","))` without this.